shakenfist/scheduler.py

```diff
@@ -239,73 +240,89 @@ def place_instance(self, instance, network, candidates=None):
             for n in copy.copy(candidates):
-                if not self._has_sufficient_cpu(log_ctx, instance.cpus, n):
+                if not self._has_sufficient_cpu(log_ctx, inst.cpus, n):
                     candidates.remove(n)
             log_ctx.with_field('candidates', candidates).info(
                 'Scheduling: have enough idle CPU')
-            instance.add_event('schedule',
-                               'Have enough idle CPU', None, str(candidates))
+            inst.add_event('schedule',
+                           'Have enough idle CPU', None, str(candidates))
             if not candidates:
                 raise exceptions.LowResourceException(
                     'No nodes with enough idle CPU')
 
             # Do we have enough idle RAM?
             for n in copy.copy(candidates):
-                if not self._has_sufficient_ram(log_ctx, instance.memory, n):
+                if not self._has_sufficient_ram(log_ctx, inst.memory, n):
                     candidates.remove(n)
             log_ctx.with_field('candidates', candidates).info(
                 'Scheduling: Have enough idle RAM')
-            instance.add_event('schedule',
-                               'Have enough idle RAM', None, str(candidates))
+            inst.add_event('schedule',
+                           'Have enough idle RAM', None, str(candidates))
             if not candidates:
                 raise exceptions.LowResourceException(
                     'No nodes with enough idle RAM')
 
             # Do we have enough idle disk?
             for n in copy.copy(candidates):
-                if not self._has_sufficient_disk(log_ctx, instance, n):
+                if not self._has_sufficient_disk(log_ctx, inst, n):
                     candidates.remove(n)
             log_ctx.with_field('candidates', candidates).info(
                 'Scheduling: Have enough idle disk')
-            instance.add_event('schedule',
-                               'Have enough idle disk', None, str(candidates))
+            inst.add_event('schedule',
+                           'Have enough idle disk', None, str(candidates))
             if not candidates:
                 raise exceptions.LowResourceException(
                     'No nodes with enough disk space')
 
+            # Calc pseudo CPU load from inst affinity
+            pseudo_load = defaultdict(int)
+            cpu_affinity = inst.affinity.get('cpu')
+            if cpu_affinity:
+                for i in instance.Instances([instance.healthy_states_filter]):
+                    if not i.tags:
+                        continue
+                    for tag, val in cpu_affinity.items():
+                        if tag in i.tags:
+                            # Use get() to allow for unplaced healthy instances
+                            pseudo_load[i.placement.get('node')] += val
+
+            inst.add_event('schedule', 'Pseudo load due inst affinity',
```

Comment:
I feel like "pseudo load due inst affinity" needs to be an english sentence which would make sense to users given its in a user facing event message.

Reply:
good point.